sql/parser/eval.go

```diff
@@ -401,14 +401,33 @@ var cmpOps = map[cmpArgs]func(Datum, Datum) (DBool, error){
 
 	cmpArgs{Like, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
 		pattern := regexp.QuoteMeta(string(right.(DString)))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Replace LIKE specific wildcards with standard wildcards
 		pattern = strings.Replace(pattern, "%", ".*", -1)
 		pattern = strings.Replace(pattern, "_", ".", -1)
-		pattern = fmt.Sprintf("^%s$", pattern)
 
 		// TODO(pmattis): Can we cache this regexp compilation?
 		re := regexp.MustCompile(pattern)
 		return DBool(re.MatchString(string(left.(DString)))), nil
 	},
+
+	cmpArgs{SimilarTo, stringType, stringType}: func(left Datum, right Datum) (DBool, error) {
+		pattern := string(right.(DString))
+		pattern = fmt.Sprintf("^%s$", pattern)
+		// Escape all periods, because they are NOT metacharacters for SQL-style regexp
+		pattern = strings.Replace(pattern, ".", "\\.", -1)
+		// Replace SQL-style regexp specific wildcards with standard regexp wildcards
+		pattern = strings.Replace(pattern, "%", ".*", -1)
+		pattern = strings.Replace(pattern, "_", ".", -1)
```

Comment:
The PostgreSQL function was really helpful in understanding the exact spec of a SQL:2008 regexp pattern. I structured a SimilarEscape function roughly off of their implementation, including adding support for a custom escape character. This could be used in the future if we ever wanted to add support for [custom escape characters](http://www.postgresql.org/docs/9.0/static/functions-matching.html#FUNCTIONS-SIMILARTO-REGEXP) in SIMILAR TO or SUBSTRING operations.

I also broke out the pattern wrapping into a separate function with support for case insensitivity.